### PR TITLE
add: support for ARRAY(ENUM) (v3)

### PR DESCRIPTION
**Pull request by sushantdhiman**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](../CONTRIBUTING.md)?

### Description of change

Add support for `ARRAY(ENUM)`

https://github.com/sequelize/sequelize/issues/1498

### CODE REVIEW

File: lib/query-interface.js [hunk 1]
```diff
@@ -153,11 +165,32 @@ QueryInterface.prototype.createTable = function(tableName, attributes, options,
...
+
       sql = self.QueryGenerator.createTableQuery(tableName, attributes, options);
 
-      return Promise.all(promises).then(function() {
-        return self.sequelize.query(sql, options);
-      });
+      return Promise.all(promises)
+        .then(function() {
+          // fetch OIDs ENUM, refresh them
+          return self.sequelize.query(
```
Comment (sushantdhiman) [inline lib/query-interface.js]: Sync will create new enum here, I couldn't find a better place to refresh oids

File: test/integration/dialects/postgres/dao.test.js [hunk 2]
```diff
@@ -393,6 +393,142 @@ if (dialect.match(/^postgres/)) {
...
+                DataTypes.ENUM('view', 'access', 'edit', 'write', 'check', 'delete')
+              )
+            });
+
+            return User.sync();
+          }).then(function() {
+            return this.sequelize.getQueryInterface().pgListEnums(User.getTableName());
+          }).then(function (enums) {
+            expect(enums).to.have.length(1);
+            expect(enums[0].enum_value).to.equal("{view,access,edit,write,check,delete}");
```
Comment (janmeier) [inline test/integration/dialects/postgres/dao.test.js]: I don't know if this order would ever be different? Maybe do a substr + split and assert on the array of items

File: test/integration/dialects/postgres/dao.test.js [hunk 3]
```diff
@@ -393,6 +393,142 @@ if (dialect.match(/^postgres/)) {
...
+          });
+        });
+
+        it('should be able to insert new record', function() {
```
Comment (janmeier) [inline test/integration/dialects/postgres/dao.test.js]: Add a test that check that you are not able to insert a value that's not part of the eunm

File: test/integration/dialects/postgres/dao.test.js (hunk 2 above)
Comment (sushantdhiman) [reply test/integration/dialects/postgres/dao.test.js]: Order wont change. I ensure you this wont turn into an intermittent failing test :D

File: test/integration/dialects/postgres/dao.test.js (hunk 3 above)
Comment (sushantdhiman) [reply test/integration/dialects/postgres/dao.test.js]: https://github.com/sequelize/sequelize/pull/8703/commits/402d51121507d9fca4be5b07d26d50961478c5f6

**Pull request merged**